ionc/ion_binary.c

```diff
@@ -535,6 +547,39 @@ iERR ion_binary_write_float_value( ION_STREAM  *pstream, double value )
     iRETURN;
 }
 
+iERR ion_binary_write_float_32_value(ION_STREAM  *pstream, float value )
+{
+    iENTER;
+    uint32_t intvalue = 0;
+    BYTE     image[ UINT_32_IMAGE_LENGTH ];
+    BYTE    *pb = &image[UINT_32_IMAGE_LENGTH - 1];
+    int      len;
+
+    ASSERT( UINT_32_IMAGE_LENGTH == 4 );                // we do depend on this here
+    ASSERT( UINT_32_IMAGE_LENGTH == sizeof(value) );    // we also depend on this here
+    ASSERT( pstream != NULL );
+
+    // this copy allows us to make endian issues just int issues
+    intvalue = *((uint32_t *)&value);
+
+    // write 8 bits at a time into the temp buffer
+    // from least to most significant, backwards
+    // that is starting from the back of the temp
+    // buffer (image) - and here we always write all 8 bytes
+    for (len = 4; len; len--) {
+        *pb-- = (BYTE)(intvalue & 0xff);
+        intvalue >>= 8;
+    }
+
+    // now write the bytes out from most to least significant
+    // to the output stream
+    ASSERT((pb + 1) == image);
+
+    IONCHECK(ion_binary_write_byte_array(pstream, image, 0, UINT_32_IMAGE_LENGTH));
+
+    iRETURN;
+}
```

Comment:
The copy-paste here might be worth the macro...

Reply:
I was going to say that I disagreed because of the number of parameters it would have to have:
* type of `intvalue`
* size of image
* expected size of image

Actually that's not so many, after all. Huh...